### PR TITLE
http: hide empty SSE events behind flag

### PR DESCRIPTION
**Pull request by jrudolph**
Refs #4017

### CODE REVIEW

File: akka-http/src/main/resources/reference.conf [hunk 1]
```diff
@@ -55,5 +55,11 @@ akka.http {
 
     # The maximum size for parsing lines of a server-sent event.
     max-line-size = 4096
+
+    # Empty events are used by the akka-http SSE infrastructure for heartbeats. Received heartbeats are ignored by default
+    # and not delivered in the user stream.
+    #
+    # This setting can be enabled to pass those empty events to the application for explicitly handling.
+    emit-empty-events = off
```
Comment (raboof) [inline akka-http/src/main/resources/reference.conf:63]: hmm, so if this is 'off' they are still emitted over the wire, just not passed to the application? That's not super clear from the name, but I'm not sure I have a better suggestion either.

File: akka-http/src/main/scala/akka/http/javadsl/settings/ServerSentEventSettings.scala [hunk 2]
```diff
@@ -27,9 +27,15 @@ abstract class ServerSentEventSettings private[akka] () { self: ServerSentEventS
    */
   def maxLineSize: Int
 
+  /**
+   * Should events with empty data field be passed to the application.
```
Comment (raboof) [inline akka-http/src/main/scala/akka/http/javadsl/settings/ServerSentEventSettings.scala:31]: :+1: to clarify that again here

**Review by raboof — APPROVED** [review]

File: akka-http/src/main/resources/reference.conf (hunk 1 above)
Comment (jrudolph) [reply akka-http/src/main/resources/reference.conf:63]: Yeah, I hoped that `emit` was a stream term vs. network terms.

Comment (jrudolph) [reply akka-http/src/main/resources/reference.conf:63]: Yeah, I hoped that `emit` was a stream term vs. network terms.

Comment (jrudolph) [reply akka-http/src/main/resources/reference.conf:63]: Github was lagging...

**Pull request merged**